Fix evaluate for boolean aura attributes

SamKacer opened the pull request:
#### Auras
* fix in combat, source unit attackable, and target unit attackable

I noticed there were issues with the "in combat" aura attribute, because any aura with it would never trigger. Looking in code I found it is because the `SkuAuras:RemoveTags` function can't handle booleans. So here is just a minor fix to make it work again.

Then I found 2 other boolean attributes with same logic and so updated them as well. Tested all 3 attributes after change and working fine.

Duugu commented:
Hm, thanks, but I don't understand that change. :)
The "in combat" attribute is perfectly working as it is.

SkuAuras:RemoveTags is working for booleans. It has this:
```
   if type(aValue) ~= "string" then
      return aValue
   end
```

SamKacer commented:
Thanks for suggestion. Went ahead and inlined that function. 

So in combat is working only in the case if the condition is "in combat equal true". If it is checking if it's false or if it uses the not equal operator instead then 
it doesn't work. If you look at the current evaluate function for in combat, if in combat is false, then it always returns false for evaluate, so that makes "not equal" not work.

Then you can also see that it hard codes the right hand side value for evaluation to always be true, so that makes "in combat equal false" not work.

The problem with `RemoveTags` is that first statement that I just deleted. If it is passed a `false	`, then it returns a nil and then later on `false == nil` will return `true`.
It can be safely deleted, since next statement that you kindly pointed out to me, will just return the value as is, if it isn't a string, so it also covers the nil case.

Then there is also issue in the function for the is and isNot operators. If either value is false, it will always return false.

So now I changed it to pass booleans instead of strings like I was doing before, but I also implemented the small fixes in `RemoveTags` and in the funcs for is and isNot operators. Retested and working as expected.

SamKacer commented:
woops, updated last comment. Meant to write `false == nil` is `false` :)

Duugu commented:
Ahh. Nice. Now it makes sense. :) Thanks!
